src/lightbox/shareImage.js

```diff
@@ -3,26 +3,31 @@ import { Platform, Share } from 'react-native';
 
 import downloadImage from './downloadImage';
 import type { Auth } from '../types';
-import ShareImageAndroid from '../nativeModules/ShareImageAndroid';
+import ShareFile from '../nativeModules/ShareFileAndroid';
 import { showToast } from '../utils/info';
 import * as api from '../api';
 import openLink from '../utils/openLink';
+import { getResource } from '../utils/url';
 
-export default async (url: string, auth: Auth) => {
-  const tempUrl = await api.tryGetFileTemporaryUrl(url, auth);
```

Comment:
I thought it was an extra step, why bother creating a temp URL when you can manage to fetch the resource regardless, I didn't see the history of the file to understand why it was introduced and assumed the initial implementation worked that way. I wasn't familiar with looking into history of a file like that and will surely use it a lot now, thanks for that!
If you would like I will revert it back to use temp URL, just give me another cue. But I still have my doubts on this, wasn't this problem in the backend rather than the fetch code? @chrisbobbe If possible, can you clarify where it won't work with directly using auth & url, where it would work using temp url.

Reply:
> If possible, can you clarify where it fails to work when using just auth & url but works using temp url.

Zulip Cloud realms on zulipchat.com use the s3 backend to store files; chat.zulip.org doesn't. So a good first step, if you'd like to investigate whether ed645fe91 went in the wrong direction, would be to experiment on a realm there. (In addition to looking for clues in its PR, #4089, and links from there.)